src/renderer/popout.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import 'renderer/css/index.css';
+
+import React, {useEffect, useState} from 'react';
+import ReactDOM from 'react-dom';
+
+import BasePage, {ErrorState} from './components/BasePage';
+import {useConfig} from './hooks/useConfig';
+import IntlProvider from './intl_provider';
+import {printVersion} from './utils';
+
+function Popout() {
+    const {config} = useConfig();
+    const [errorState, setErrorState] = useState<ErrorState | undefined>(undefined);
+    const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+    const [errorUrl, setErrorUrl] = useState<string | undefined>(undefined);
+    const [title, setTitle] = useState<string | undefined>(undefined);
+    const [viewId, setViewId] = useState<string | undefined>(undefined);
+
+    useEffect(() => {
+        printVersion();
+
+        // Deny drag&drop navigation in mainWindow.
+        // Drag&drop is allowed in webview of index.html.
+        document.addEventListener('dragover', (event) => event.preventDefault());
+        document.addEventListener('drop', (event) => event.preventDefault());
+
+        window.desktop.onLoadFailed((_, err, loadUrl) => {
+            setErrorState(ErrorState.FAILED);
+            setErrorMessage(err);
+            setErrorUrl(loadUrl);
+        });
+
+        window.desktop.onLoadIncompatibleServer((_, loadUrl) => {
+            setErrorState(ErrorState.INCOMPATIBLE);
+            setErrorUrl(loadUrl);
+        });
+
+        window.desktop.onUpdatePopoutTitle((viewId, title) => {
+            setTitle(title);
+            setViewId(viewId);
+        });
```

Comment:
Shouldn't we de-register all those callbacks?

Reply:
This needs to be added and redone for all internal listeners. Since the assumption was that they've never registered more than once, and destroyed when the window closes, we didn't set that up for internal stuff.

The externalAPI does this though. I'll look at it at a later date.